lcdb/reporting.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+""" Reporting System """
+
+import os
+import sys
+import base64
+from textwrap import dedent
+
+import markdown
+import jinja2
+
+TEMPLATES_DIR = os.path.join(os.path.dirname(__file__), 'templates')
+loader = jinja2.FileSystemLoader(TEMPLATES_DIR)
+ENV = jinja2.Environment(loader=loader)
+
+def report(string, **kwargs):
+    """ Build a report """
+    md = markdown.Markdown(output_format='html5')
+    tp = ENV.get_template('base.html')
+    return tp.render(content=md.convert(dedent(string).format(**kwargs)))
+
```

Comment:
Just to clarify: in this system, each logical chunk of a report will have its own rule to create HTML.  The snakemake infrastructure will identify which rules will need to run and will thereforel be used to determine whether a chunk needs to be built. The final report rule in turn will essentially stitch together the various chunks into one final HTML report. Is that right?